src/test/java/uk/gov/ida/eventemitter/EventEmitterTest.java

```diff
@@ -1,68 +1,112 @@
 
+@RunWith(MockitoJUnitRunner.class)
 public class EventEmitterTest {
 
-    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private static final String ENCRYPTED_EVENT = "encrypted event";
+    private static final UUID ID = UUID.randomUUID();
+    private static final String TIMESTAMP = "2018-02-06T14:37:55.467Z";
+    private static final String EVENT_TYPE = "Error Event";
 
     private EventEmitter eventEmitter;
     private EventEmitter eventEmitterWithoutRecordingSystem;
+    private TestEvent event;
+
+    @Mock
+    private Encrypter encrypter;
+
+    @Mock
+    private SqsClient sqsClient;
 
     @Before
     public void setUp() {
-        System.setOut(new PrintStream(outContent));
+        when(encrypter.encrypt(any(Event.class))).thenReturn(ENCRYPTED_EVENT);
 
-        eventEmitter = new EventEmitter(true);
-        eventEmitterWithoutRecordingSystem = new EventEmitter(false);
-    }
+        eventEmitter = new EventEmitter(true, encrypter, sqsClient);
+        eventEmitterWithoutRecordingSystem = new EventEmitter(false, encrypter, sqsClient);
 
-    @After
-    public void tearDown() {
-        System.setOut(System.out);
+        event = new TestEvent(ID, DateTime.parse(TIMESTAMP), EVENT_TYPE);
     }
 
     @Test
-    public void recordMethodShouldWriteEventDetailsToSystemOut() {
-        final UUID id = UUID.randomUUID();
-        final String timestamp = "2018-02-06T14:37:55.467Z";
-        final String eventType = "Error Event";
-        final TestEvent event = new TestEvent(id, DateTime.parse(timestamp), eventType);
-
-        eventEmitter.record(event);
-
-        assertThat(outContent.toString())
-            .isEqualTo(String.format("Event ID: %s, Timestamp: %s, Event Type: %s\n", id, timestamp, eventType));
+    public void shouldWriteEventDetailsToSystemOut() throws IOException {
+        try (ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+             PrintStream printStream = new PrintStream(outContent)) {
+            System.setOut(printStream);
+            eventEmitter.record(event);
+            System.setOut(System.out);
+
+            assertThat(outContent.toString()).containsOnlyOnce(String.format(
+                "Event ID: %s, Timestamp: %s, Event Type: %s, Event String: %s\n",
+                ID,
+                TIMESTAMP,
+                EVENT_TYPE,
+                ENCRYPTED_EVENT
+            ));
+        }
     }
 
     @Test
-    public void recordMethodShouldNotWriteEventDetailsToSystemOut() {
-        final UUID id = UUID.randomUUID();
-        final String timestamp = "2018-02-06T14:37:55.467Z";
-        final String eventType = "Error Event";
-        final TestEvent event = new TestEvent(id, DateTime.parse(timestamp), eventType);
-
-        eventEmitterWithoutRecordingSystem.record(event);
-
-        assertThat(outContent.toString()).isEmpty();
+    public void shouldNotWriteEventDetailsToSystemOut() throws IOException {
```

Comment:
this test name seems to contradict the one above. Could you add a little more context to each of them?

Reply:
I have made the changes.